[WPE] WPE Platform: Focus toplevel view on interaction

Scony opened the pull request:
#### 1b5eef41b5f33d748470e7b34bcbc87bc29f432a
<pre>
[WPE] WPE Platform: Focus toplevel view on interaction
<a href="https://bugs.webkit.org/show_bug.cgi?id=293876">https://bugs.webkit.org/show_bug.cgi?id=293876</a>

Reviewed by Carlos Garcia Campos.

This change makes user interactions trigger focus.

* Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.cpp:

Canonical link: <a href="https://commits.webkit.org/295911@main">https://commits.webkit.org/295911@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ccccef1c6b83cd2aaf5f9140aacf5f81835c07e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106402 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26151 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16549 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111603 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56998 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26818 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34654 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80817 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109406 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21266 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96029 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61145 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20746 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56438 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90596 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14163 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114462 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33540 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24732 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89886 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33904 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92258 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89589 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34484 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12308 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29147 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17262 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33465 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38877 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33211 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36564 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34809 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/bde64373589c4752df4e3e0a680b984235179a06. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46187

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.cpp:
```diff
@@ -166,7 +166,12 @@ const struct wl_pointer_listener WaylandSeat::s_pointerListener = {
         seat.m_pointer.time = time;
 
-        GRefPtr<WPEView> view = wpeToplevelWaylandGetVisibleViewUnderPointer(seat.m_pointer.toplevel.get());
+        auto* toplevelWayland = seat.m_pointer.toplevel.get();
+        wpeToplevelWaylandSetIsFocused(toplevelWayland, true);
+        if (GRefPtr<WPEView> focusedView = wpeToplevelWaylandGetVisibleFocusedView(toplevelWayland))
+            wpe_view_focus_in(focusedView.get());
+
+        GRefPtr<WPEView> view = wpeToplevelWaylandGetVisibleViewUnderPointer(toplevelWayland);
```
I think the focus on button press/touch should not be platform specific, it should be done by WebKit in WPEWebViewPlatform like gtk does in WebKitWebViewBase.

Scony committed:
[WPE] WPE Platform: Focus toplevel view on interaction
https://bugs.webkit.org/show_bug.cgi?id=293876

Reviewed by Carlos Garcia Campos.

This change makes user interactions trigger focus.

* Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.cpp:

Canonical link: https://commits.webkit.org/295911@main

webkit-commit-queue commented:
Committed 295911@main (1b5eef41b5f3): <https://commits.webkit.org/295911@main>

Reviewed commits have been landed. Closing PR #46187 and removing active labels.
